### ISSUE TITLE
Use icons as field labels in Datagrid

### ISSUE THREAD

**Issue by deluan** [marmelab/react-admin#4848]
**Is your feature request related to a problem? Please describe.**
I want to show an icon as a column header in Datagrid, like this:
<img width="843" alt="Screen Shot 2020-05-24 at 10 00 36 PM" src="https://user-images.githubusercontent.com/331353/82771476-338d9600-9e0a-11ea-894c-1eb50a62faba.png">

I can achieve this with: 
```jsx
label={<StarBorderIcon fontSize={'small'} className={classes.columnIcon} />}
```
The problem is that the `FieldTitle` component always tries to translate the field's `label`, and gives this error in the console:
```console
Warning: Missing translation for key: "[object Object]" 
    at FieldTitle (http://localhost:4533/static/js/0.chunk.js:106580:21)
    at pure(FieldTitle) (http://localhost:4533/static/js/0.chunk.js:187826:9)
    at th
    at TableCell (http://localhost:4533/static/js/0.chunk.js:27368:28)
    at WithStyles (http://localhost:4533/static/js/0.chunk.js:41841:31)
    at DatagridHeaderCell (http://localhost:4533/static/js/0.chunk.js:131626:25)
...
```

**Describe the solution you'd like**
Maybe we could have a new property in the Field that accepts a component as label, instead of using the `label` itself (`iconLabel`?).

**Describe alternatives you've considered**
Can't think of anything else.

**Comment by fzaninotto** [comment]
Definitely useful! I actually need it, too. Marking it as a feature request, and working on an implementation.

**Comment by deluan** [comment]
By the way, it would be nice if the implementation aligns the icon for you. To get the correct position, I had to add the following style to the icon:
```
  columnIcon: {
    marginLeft: '3px',
    verticalAlign: 'text-top',
  },
```

**Comment by deluan** [comment]
I just though about a better solution: the `FieldTitle` component could check for the type of the `label` field, and only tries to translate it if it is a string

**Comment by fzaninotto** [comment]
@deluan That's what I did, see attached PR.

**Comment by fzaninotto** [comment]
Fixed by #4852
